### ISSUE TITLE
Heapster InfluxDB sink fails silently with heapster-influxdb:v0.13.0

### ISSUE THREAD

**Issue by jmn** [kubernetes/heapster#1482]
commit 5bf0e3e9144c2b0d75e56a522da046e449060543 seems to break the heapster/influxdb/grafana example. 

Specifically changing image:
        image: kubernetes/heapster_influxdb:v0.6
+        image: gcr.io/google_containers/heapster-influxdb:v0.13.0

Running influxdb v0.13.0 will cause the Heapster InfluxDB sink to fail (silently because there does not appear to be any error logging facilities.)

**Comment by whereisaaron** [comment]
@jmn the influxdb version has since been updated to v1.1.1. But the example is still failing for me too. heapster 1.3.0-beta.0 appears to be failing to write to influxdb v1.1.1

Failed the same way with heapster 1.2.0.

The influxdb log shows the create database went okay, but then 500 errors for POSTs from heapster:
```
[httpd] 10.2.67.16 - root [25/Jan/2017:06:18:05 +0000] "GET /query?db=&q=CREATE+DATABASE+k8s HTTP/1.1" 200 163 "-" "heapster/v1.3.0-beta.0" 08387ab6-e2c6-11e6-80b9-000000000000 666
[httpd] 10.2.67.16 - root [25/Jan/2017:06:18:05 +0000] "POST /write?consistency=&db=k8s&precision=&rp=default HTTP/1.1" 500 72 "-" "heapster/v1.3.0-beta.0" 08406030-e2c6-11e6-80ba-000000000000 2389
[httpd] 10.2.67.16 - root [25/Jan/2017:06:18:05 +0000] "GET /ping HTTP/1.1" 204 0 "-" "heapster/v1.3.0-beta.0" 0840ccfd-e2c6-11e6-80bb-000000000000 60
[httpd] 10.2.67.16 - root [25/Jan/2017:06:19:05 +0000] "POST /write?consistency=&db=k8s&precision=&rp=default HTTP/1.1" 500 72 "-" "heapster/v1.3.0-beta.0" 2bfe3472-e2c6-11e6-80ed-000000000000 3289
[httpd] 10.2.67.16 - root [25/Jan/2017:06:19:05 +0000] "GET /ping HTTP/1.1" 204 0 "-" "heapster/v1.3.0-beta.0" 2bfec387-e2c6-11e6-80ee-000000000000 11
```

The heapster log seems happy at start-up:
```
I0125 06:17:01.230742       1 heapster.go:71] /heapster --source=kubernetes.summary_api:'' --sink=influxdb:http://monitoring-influxdb:8086
I0125 06:17:01.230818       1 heapster.go:72] Heapster version v1.3.0-beta.0
I0125 06:17:01.231026       1 configs.go:60] Using Kubernetes client with master "https://10.3.0.1:443" and version v1
I0125 06:17:01.231065       1 configs.go:61] Using kubelet port 10255
I0125 06:17:01.233219       1 influxdb.go:231] created influxdb sink with options: host:monitoring-influxdb:8086 user:root db:k8s
I0125 06:17:01.233236       1 heapster.go:193] Starting with InfluxDB Sink
I0125 06:17:01.233239       1 heapster.go:193] Starting with Metric Sink
I0125 06:17:01.525442       1 heapster.go:105] Starting heapster on port 8082
I0125 06:18:05.070935       1 influxdb.go:209] Created database "k8s" on influxDB server at "monitoring-influxdb:8086"
```

**Comment by whereisaaron** [comment]
This bug appears to have been discussed and resolved in #1474 (closed already so didn't spot it).

**Comment by jmn** [comment]
I think I provided the wrong version number of the InfluxDB in the issue report here. I think this report is actually regarding gcr.io/google_containers/heapster-influxdb-amd64:v1.1.1 But at this point I am not actually sure. I deployed image: kubernetes/heapster_influxdb:v0.6 and it works on my cluster for now.

@whereisaaron Can you confirm a successful deployment of the examples as they are currently provided?

**Comment by whereisaaron** [comment]
The [example](https://github.com/kubernetes/heapster/blob/master/docs/influxdb.md) failed for me to with the same 500 errors. This fix worked for me too.

I wasn't sure where to get the influx CLI client from, found it here: https://www.influxdata.com/downloads/

I made the following changes and it started working.
```
InfluxDB shell version: 1.2.0
> show databases
name: databases
name
----
_internal
k8s

> use k8s
Using database k8s

> show retention policies
name    duration shardGroupDuration replicaN default
----    -------- ------------------ -------- -------
autogen 0s       168h0m0s           1        true

> CREATE RETENTION POLICY "default" ON k8s DURATION 0d REPLICATION 1 DEFAULT

> show retention policies
name    duration shardGroupDuration replicaN default
----    -------- ------------------ -------- -------
autogen 0s       168h0m0s           1        false
default 0s       168h0m0s           1        true
```

**Comment by lilnate22** [comment]
i am getting same issue (ish)

```
: getsockopt: connection timed out
W0129 15:59:25.064674       1 manager.go:119] Failed to push data to sink: InfluxDB Sink
W0129 16:00:25.039653       1 manager.go:119] Failed to push data to sink: InfluxDB Sink
E0129 16:00:26.729637       1 influxdb.go:150] Failed to create infuxdb: failed to ping InfluxDB server at "monitoring-influxdb:8086" - Get http://monitoring-influxdb:8086/ping: dial tcp 10.103.30.121:8086: getsockopt: connection timed out
```

from my heapster... later on down the line:

```
0129 16:29:30.957206       1 listers.go:68] can not retrieve list of objects using index : object has no meta: object does not implement the Object interfaces
W0129 16:30:00.942962       1 listers.go:68] can not retrieve list of objects using index : object has no meta: object does not implement the Object interfaces
W0129 16:30:30.949341       1 listers.go:68] can not retrieve list of objects using index : object has no meta: object does not implement the Object interfaces
W0129 16:30:30.970448       1 listers.go:68] can not retrieve list of objects using index : object has no meta: object does not implement the Object interfaces
W0129 16:30:31.017071       1 listers.go:68] can not retrieve list of objects using index : object has no meta: object does not implement the Object interfaces
```

**Comment by KarolKraskiewicz** [comment]
Hi Everyone,

It seems the bug described here has been fixed and changes have been released within the new container: gcr.io/google_containers/heapster-amd64:v1.3.0-beta.1

I verified, that last version of heapster/influxdb/grafana configuration:
https://github.com/kubernetes/heapster/tree/master/deploy/kube-config/influxdb
which is using the new container, is working properly.

@jmn - is there anything else missing here from your perspective?
@lilnate22 - are you still experiencing issues described above? If yes, which version of heapster / influxdb / grafana images are you using?
